flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -426,4 +431,17 @@ protected boolean flinkVersionChanged(SPEC oldSpec, SPEC newSpec) {
         }
         return false;
     }
+
+    private void setOwnerReference(CR owner, Configuration deployConfig) {
+        final Map<String, String> ownerReference =
+                Map.of(
+                        "apiVersion", owner.getApiVersion(),
+                        "kind", owner.getKind(),
+                        "name", owner.getMetadata().getName(),
+                        "uid", owner.getMetadata().getUid(),
+                        "blockOwnerDeletion", "false",
+                        "controller", "false");
+        deployConfig.set(
+                KubernetesConfigOptions.JOB_MANAGER_OWNER_REFERENCE, List.of(ownerReference));
+    }
```

Comment:
Feels like the `FlinkConfigBuilder` might be a better place for this, then we can be sure that it's universally set in a single place.

Reply:
Or the `getDeployConfig` method of the `FlinkConfigManager` it might be easier to add there :)

Reply:
@gyfora 
I tried it but, the problem is with the apiVersion and kind

Is there a good way to add the apiVersion and kind of a CustomResource, if not in the `AbstractFlinkResourceReconciler`?

Reply:
We could probably change the information passed to getDeployConfig if necessary. Pass the whole CR + spec instead of meta + spec

Reply:
but I dont have a better idea